torch/csrc/jit/codegen/cuda/runtime/helpers.cu

```diff
@@ -620,3 +620,64 @@ __device__ __bfloat print_impl(const char* name, __bfloat value) {
 }
 
 #define print(...) print_impl(#__VA_ARGS__, (__VA_ARGS__))
+
+//! Utility to print out bank conflicts given the address from each thread.
+__device__ void checkBankConflict(
+    size_t address,
+    int vec_word_in_byte,
+    int call_id) {
+  // Calculate number of threads in each phase that could result in
+  //  bank conflict.
+  int phase_size = vec_word_in_byte == 16 ? 8 : vec_word_in_byte == 8 ? 16 : 32;
+
+  // Initialize address space to populate
+  size_t address_warp[32];
+  for (int i = 0; i < 32; i++) {
+    address_warp[i] = 0;
+  }
+
+  // Calculate lane_id
+  int lane_id = 0;
+
+  // Read laneid register directly with ptx.
+  asm("mov.u32 %0, %%laneid;" : "=r"(lane_id) :);
+
+  // Populate address within each phase:
+  // for more info on shared memory access phase see page 66 of:
+  // https://on-demand.gputechconf.com/gtc/2018/presentation/s81006-volta-architecture-and-performance-optimization.pdf
+  for (int i = 0; i < phase_size; i++) {
+    size_t shfl_value = 0;
+    shfl_value = __shfl_down_sync(__activemask(), address, i);
+
+    if (__activemask() & (1 << (lane_id + i))) {
+      address_warp[i] = shfl_value;
+    }
+  }
+
+  // Check bank conflict at phase head, i.e. the first lane in each phase.
+  if (lane_id % phase_size == 0) {
+    for (int i = 0; i < phase_size; i++) {
+      for (int j = 0; j < phase_size; j++) {
+        if (address_warp[i] != 0 && address_warp[j] != 0 && i < j) {
+          size_t bank_idx_i = (address_warp[i] % 128) /
+              vec_word_in_byte; // 128Byte per memory row.
+          size_t bank_idx_j = (address_warp[j] % 128) /
+              vec_word_in_byte; // 128Byte per memory row.
+
+          if (bank_idx_i == bank_idx_j && address_warp[i] != address_warp[j]) {
```

Comment:
I don't think this is correct. See picture:
![Screenshot_20220929_121130](https://user-images.githubusercontent.com/1032377/193121762-9c446f85-d980-4e4f-a05d-093301ea700b.jpg)
![Screenshot_20220929_121229](https://user-images.githubusercontent.com/1032377/193121803-6ede9058-943d-40e9-9ef6-f1e0612e23ff.jpg)